src/base/static/js/views/app-view.js

```diff
@@ -846,52 +699,100 @@ var Shareabouts = Shareabouts || {};
           }
         }
 
-        // Focus the one we're looking
         model.trigger('focus');
-
-        if (model.get("story")) {
-          if (!model.get("story").spotlight) $("#spotlight-place-mask").remove();
-          self.isStoryActive = true;
-          self.setStoryLayerVisibility(model);
-        } else if (self.isStoryActive) {
+        
+        if (!model.get("story") && self.isStoryActive) {
           self.isStoryActive = false;
           self.restoreDefaultLayerVisibility();
-        } else {
-          self.isStoryActive = false;
         }
       };
 
-      onPlaceNotFound = function() {
+      onNotFound = function() {
         self.options.router.navigate('/');
+        return;
       };
 
-      // If we get a PlaceModel then show it immediately.
-      if (model instanceof S.PlaceModel) {
-        onPlaceFound(model);
-        return;
-      }
+      searchLoadedCollections = function(collections, property, type) {
```

Comment:
Maybe we can abstract `searchLoadedCollections` and `bindCollectionsToListeners` into a separate util method. Maybe we can also call it something more useful, like `findPlaceFromCollections` and have it take a callback that calls `onFound` when the place is found.